cmd/hubagent/options/azureoptions.go

```diff
@@ -0,0 +1,31 @@
+//
+//Copyright (c) Microsoft Corporation.
+//Licensed under the MIT license.
+
+package options
+
+import (
+	"flag"
+)
+
+// AzurePropertyCheckerOptions holds the options for the Azure property checker.
+type AzurePropertyCheckerOptions struct {
+	// IsEnabled indicates whether the Azure property checker is enabled.
+	IsEnabled bool
+	// ComputeServiceAddressWithBasePath is the address of the Azure compute service with base path.
+	ComputeServiceAddressWithBasePath string
+}
+
+// NewAzurePropertyCheckerOptions creates a new AzurePropertyCheckerOptions with default values.
+func NewAzurePropertyCheckerOptions() AzurePropertyCheckerOptions {
+	return AzurePropertyCheckerOptions{
+		IsEnabled:                         false,
+		ComputeServiceAddressWithBasePath: "http://localhost:8421/compute",
+	}
+}
+
+// AddFlags adds flags for Azure Property Checker options to the given FlagSet.
+func (o AzurePropertyCheckerOptions) AddFlags(flags *flag.FlagSet) {
```

Comment:
I'm not sure if we want to make this a pointer. cc @zhiying-lin like RatelimiterOpts: https://github.com/kubefleet-dev/kubefleet/blob/main/cmd/hubagent/options/ratelimit.go#L43